### ISSUE TITLE
Rules do not respect artifact-specific overrides

### ISSUE THREAD

**Issue by OliverO2** [jmfayard/refreshVersions#363]
- [x] I have read the [guidelines for submitting issues](https://jmfayard.github.io/refreshVersions/contributing/submitting-issues/)

## 🐛 Describe the bug

Currently, it appears that a rule like `kotlin(x)-version-alias-rules.txt` will always override the version given for a specific artifact matching that rule. (Additionally, the matching seems to be too greedy, see note 1 below.)

## ⚠️ Current behavior

[Kotlin/kotlinx-benchmark](https://github.com/Kotlin/kotlinx-benchmark) comes with its own plugin and uses a versioning scheme independent of generic Kotlin versioning. The current version is `0.3.1`. Currently, this cannot be used as documented.

## ✅ Expected behavior

The generic Kotlin version should never override an artifact-specific one.

##### Example 1

The artifact-specific version given in `versions.properties` is ignored.
* `build.gradle.kts`:
    ```kotlin
    plugins {
        kotlin("multiplatform")
        id("org.jetbrains.kotlinx.benchmark")
    }
    ```
* `versions.properties`:
    ```kotlin
    version.kotlin=1.4.32
    
    plugin.org.jetbrains.kotlinx.benchmark = 0.3.1
    ```
* Result:
    ```
    Plugin [id: 'org.jetbrains.kotlinx.benchmark', version: '1.4.32'] was not found in any of the following sources:
    ```

##### Example 2
Deliberately introducing a different spelling (`kotylinx`) for demonstration purposes. Now the artifact-specific version is picked up. 
* `build.gradle.kts`:
    ```kotlin
    plugins {
        kotlin("multiplatform")
        id("org.jetbrains.kotylinx.benchmark")
    }
    ```
* `versions.properties`:
    ```kotlin
    version.kotlin=1.4.32
    
    plugin.org.jetbrains.kotylinx.benchmark = 0.3.1
    ```
* Result:
    ```
    Plugin [id: 'org.jetbrains.kotylinx.benchmark', version: '0.3.1'] was not found in any of the following sources:
    ```

## Workaround

The buildscript approach works:

```kotlin
buildscript {
    dependencies {
        classpath("org.jetbrains.kotlinx:kotlinx-benchmark-plugin:_")
    }
}

apply(plugin = "org.jetbrains.kotlinx.benchmark")
```

## Notes

1. Renaming the artifact's group to `org.jetbrains.kotlinyx.benchmark` would still match the Kotlin rule, though it does neither end with `kotlin` nor with `kotlinx`:
    ```
    Plugin [id: 'org.jetbrains.kotlinyx.benchmark', version: '1.4.32'] was not found in any of the following sources:
    ```

1. An artifact-specific version may also be required if the generic version does not work as required.

**Comment by LouisCAD** [comment]
Yikes, using `startsWith` is not a good thing here.

Thanks for the good report, will fix ASAP.

**Comment by LouisCAD** [comment]
Fixed in 0.10.1, released last week.
